### PR TITLE
chore(Scripts/GruulsLair): modernise boss scripts + make timers more accurate

### PR DESCRIPTION
**Pull request by elthehablo**
## Changes Proposed:
-  using TaskScheduler on both bosses instead of events
-  using sniffs supplied by Gultask to try to improve the timers

## Issues Addressed:

- Closes 

## SOURCE:

see my research for the timers [here](https://github.com/elthehablo/GeneralACResearch/blob/main/gruulfindings.md)

## Tests Performed:

- tested in-game
- 

## How to Test the Changes:

1. go to Gruul's lair to checkout the slightly improved timers

## Known Issues and TODO List:

- [ ] timers are not completely final. some are still likely not entirely correct yet.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Outland/GruulsLair/boss_gruul.cpp [hunk 1]
```diff
@@ -58,46 +58,77 @@ enum Spells
...
+            context.Repeat(22s);
+        }).Schedule(10s, [this](TaskContext context)
+        {
+            if (Unit* target = SelectTarget(SelectTargetMethod::MaxThreat, 1, 5.0f))
+            {
+                me->CastSpell(target, SPELL_HURTFUL_STRIKE, false);
+            }
+            else
+            {
+                me->CastSpell(me->GetVictim(), SPELL_HURTFUL_STRIKE, false);
```
Comment (Nyeriah) [inline src/server/scripts/Outland/GruulsLair/boss_gruul.cpp:102]: DoCastVictim

File: src/server/scripts/Outland/GruulsLair/boss_gruul.cpp [hunk 2]
```diff
@@ -58,46 +58,77 @@ enum Spells
...
+            context.Repeat(_caveInTimer);
+        }).Schedule(20s, [this](TaskContext context)
+        {
+            DoCastSelf(SPELL_REVERBERATION);
+            context.Repeat(22s);
+        }).Schedule(10s, [this](TaskContext context)
+        {
+            if (Unit* target = SelectTarget(SelectTargetMethod::MaxThreat, 1, 5.0f))
+            {
+                me->CastSpell(target, SPELL_HURTFUL_STRIKE, false);
```
Comment (Nyeriah) [inline src/server/scripts/Outland/GruulsLair/boss_gruul.cpp:98]: DoCast(target, SPELL_HURTFUL_STRIKE);

File: src/server/scripts/Outland/GruulsLair/boss_gruul.cpp [hunk 3]
```diff
@@ -325,4 +312,4 @@ void AddSC_boss_gruul()
     RegisterSpellScript(spell_gruul_ground_slam_trigger);
     RegisterSpellScript(spell_gruul_shatter);
     RegisterSpellScript(spell_gruul_shatter_effect);
-}
+}
```
Comment (Nyeriah) [inline src/server/scripts/Outland/GruulsLair/boss_gruul.cpp:315]: I think it needs a blank line at the end of the file, seems to be complaining about it

File: src/server/scripts/Outland/GruulsLair/boss_high_king_maulgar.cpp [hunk 4]
```diff
@@ -468,4 +451,4 @@ void AddSC_boss_high_king_maulgar()
     RegisterGruulsLairAI(boss_blindeye_the_seer);
     RegisterGruulsLairAI(boss_olm_the_summoner);
     RegisterGruulsLairAI(boss_krosh_firehand);
-}
+}
```
Comment (Nyeriah) [inline src/server/scripts/Outland/GruulsLair/boss_high_king_maulgar.cpp:454]: same here

**Comment by elthehablo** [comment]
freaking blanklines

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**